packages/frontend/src/lib/pnl.ts

```diff
@@ -1,21 +1,76 @@
 import BigNumber from 'bignumber.js'
 
+import { VaultHistory_vaultHistories } from '../queries/squeeth/__generated__/VaultHistory'
+import { toTokenAmount } from '@utils/calculations'
+
 type ShortPnLParams = {
   wethAmount: BigNumber
   buyQuote: BigNumber
   ethPrice: BigNumber
+  ethCollateralPnl: BigNumber
 }
 
-export function calcUnrealizedPnl({ wethAmount, buyQuote, ethPrice }: ShortPnLParams) {
+export function calcUnrealizedPnl({ wethAmount, buyQuote, ethPrice, ethCollateralPnl }: ShortPnLParams) {
   if (
     wethAmount.isEqualTo(0) ||
     !wethAmount.isFinite() ||
     buyQuote.isEqualTo(0) ||
     !buyQuote.isFinite() ||
     ethPrice.isEqualTo(0) ||
-    !ethPrice.isFinite()
+    !ethPrice.isFinite() ||
+    ethCollateralPnl.isEqualTo(0) ||
+    !ethCollateralPnl.isFinite()
   ) {
     return new BigNumber(0)
   }
-  return wethAmount.minus(buyQuote).multipliedBy(ethPrice)
+
+  return wethAmount.minus(buyQuote).multipliedBy(ethPrice).plus(ethCollateralPnl)
+}
+
+type ShortGainParams = {
+  shortUnrealizedPNL: BigNumber
+  usdAmount: BigNumber
+  wethAmount: BigNumber
+  ethPrice: BigNumber
+}
+
+export function calcShortGain({ shortUnrealizedPNL, usdAmount, wethAmount, ethPrice }: ShortGainParams) {
+  if (wethAmount.isEqualTo(0) || shortUnrealizedPNL.isEqualTo(0) || ethPrice.isEqualTo(0) || usdAmount.isEqualTo(0)) {
+    return new BigNumber(0)
+  }
+  return shortUnrealizedPNL.div(usdAmount.plus(wethAmount.times(ethPrice).absoluteValue())).times(100)
+}
+export function calcLongUnrealizedPnl({
+  sellQuote,
+  wethAmount,
+  ethPrice,
+}: {
+  sellQuote: BigNumber
+  wethAmount: BigNumber
+  ethPrice: BigNumber
+}) {
+  return {
+    usdValue: sellQuote.minus(wethAmount.abs()).multipliedBy(ethPrice),
+    ethValue: sellQuote.minus(wethAmount.abs()),
+  }
+}
+
+export function calcETHCollateralPnl(
+  data: VaultHistory_vaultHistories[] | undefined,
+  ethPriceMap: { [key: string]: number },
+  ethPrice: BigNumber,
+) {
+  return data?.length
+    ? data?.reduce((acc: BigNumber, curr: VaultHistory_vaultHistories) => {
+        const time = new Date(Number(curr.timestamp) * 1000).setUTCHours(0, 0, 0) / 1000
+        if (curr.action === 'DEPOSIT_COLLAT') {
```

Comment:
Just wondering if user uses MINT_SHORT to open a short position, do they also trigger the 'DEPOSIT_COLLAT' in vault history?

Reply:
Also if user uses WITHDRAW_COLLAT, the withdrawal amount of the collat shld be calculated in realized pnl instead of unrealized?

Also have a same q for BURN_SHORT, if user uses MINT_SHORT to open a short position, do they also trigger the 'WITHDRAW_COLLAT' in vault history?

Reply:
Besides, If the user gets liquidated, shld we also take the liquidated amount to get the new debt & collat and get the new realized & unrealized pnl? Does It depend on if it's a full liquidation or partial liquidation?

Reply:
> Just wondering if user uses MINT_SHORT to open a short position, do they also trigger the 'DEPOSIT_COLLAT' in vault history?
> 
> Also have a same q for BURN_SHORT, if user uses MINT_SHORT to open a short position, do they also trigger the 'WITHDRAW_COLLAT' in vault history?

Yes it does you can actually test these by performing these operations and going on the subgraph to query vault history of your vault

Reply:
> Besides, If the user gets liquidated, shld we also take the liquidated amount into account to get the new debt & collat and get the new realized & unrealized pnl? Does It depend on if it's a full liquidation or partial liquidation?

I don't know the answer to this.  Cc @alexisgauba @KMKoushik

Reply:
Full liquidation we don't need to take into account because that closes out their whole position. Partial liquidation would be good to take into account but imo doesn't need to be in this PR, can be later

Reply:
Liquidations are tough we can add it later